app/app.py

```diff
@@ -78,9 +87,16 @@ def find_recipe_in_url_window_algo_based():
     instructions = request.form['instructions'].lower() == "true"
     ingredients = request.form['ingredients'].lower() == "true"
     try:
+        if stored_recipes_dao.url_in_db(url):
+            return jsonify(stored_recipes_dao.get_recipe_from_db(url, ingredients, instructions))
         all_text, title = text_extractor.get_all_text_from_url(url=url)
         ingred_paragraph, instr_paragraph = window_key_word_based_algo.extract(ingredients, instructions, all_text)
-        return create_json_response(ingred_paragraph, instr_paragraph, title, url)
+        response = create_json_response(ingred_paragraph, instr_paragraph, title, url)
+
+        thread = threading.Thread(target=stored_recipes_dao.add_recipe_to_db, kwargs={"response": response})
+        thread.start()
```

Comment:
I love your initiative here, but this is risky. 
it means that you will allocate 1 Thread per connection.
in  "rush hour' the server will collapse. 
either use correctly thread pool or just keep it simple (anyway it will happen only once per recipe)